source/production/F0.Minesweeper.Logic/LocationShuffler/GuidLocationShuffler.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using F0.Minesweeper.Logic.Abstractions;
+
+namespace F0.Minesweeper.Logic.LocationShuffler
+{
+	internal class GuidLocationShuffler : ILocationShuffler
+	{
+		private static ILocationShuffler? instance;
+
+		private GuidLocationShuffler() { }
+
+		internal static ILocationShuffler Instance
+		{
+			get
+			{
+				instance ??= new GuidLocationShuffler();
+				return instance;
+			}
+		}
+
+		IReadOnlyCollection<Location> ILocationShuffler.ShuffleAndTake(IEnumerable<Location> allLocations, int count)
+		{
+			_ = allLocations ?? throw new ArgumentNullException(nameof(allLocations));
+
+			var allLocationsCount = allLocations.Count();
+			if (count > allLocationsCount)
+			{
+				throw new ArgumentOutOfRangeException(nameof(count), count, $"The take count should not be greater than the number of elements in {nameof(allLocations)}. Element count: {allLocationsCount}");
+			}
+
+			return allLocations
```

Comment:
Or we could also perform lines 33 to 36 first and perform a length check afterwards on the array. 
Because from my understanding `Take` doesn't throw in case you provide a count higher than the `Count()` of the enumerable but rather just returns the complete enumerable. 

With this we could simply check if the `Length` of the array is the same as the provided `count` input and if it's not then we still can throw the `ArgumentOutOfRangeException`

Reply:
Done ❤️